codalab/worker_manager/slurm_batch_worker_manager.py

```diff
@@ -0,0 +1,268 @@
+import argparse
+import logging
+import uuid
+import subprocess
+import getpass
+from pathlib import Path
+import random
+import re
+
+from .worker_manager import WorkerManager, WorkerJob
+
+logger = logging.getLogger(__name__)
+
+
+class SlurmBatchWorkerManager(WorkerManager):
+    NAME = 'slurm-batch'
+    DESCRIPTION = 'Worker manager for submitting jobs using Slurm Batch'
+
+    SRUN_COMMAND = 'srun'
+    SBATCH_COMMAND = 'sbatch'
+    SBATCH_PREFIX = '#SBATCH'
+    SQUEUE_COMMAND = 'squeue'
+    SBATCH_COMMAND_RETRUN_REGEX = re.compile(r'^Submitted batch job (\d+)$')
+
+    @staticmethod
+    def add_arguments_to_subparser(subparser):
+        subparser.add_argument(
+            '--job-name',
+            type=str,
+            default='codalab-slurm-worker',
+            help='Name for the job definitions that will be generated by this worker manager',
+        )
+        subparser.add_argument(
+            '--nodelist', type=str, default='', help='The worker node to run jobs in'
```

Comment:
Does the default of `''` work? i.e., if you set `slurm_args['nodelist'] = ''` below, is the `nodelist` all nodes, or is the nodelist no nodes? Maybe consider removing the default here, and, in `create_slurm_args`, only add the nodelist argument `if self.args.nodelist`?

Reply:
Yes, I tried that. Default `--nodelist = ''` will just randomly give you a partition.